In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_excel(r'..\task_3.xlsx')

In [3]:
df


,Месяц,Год,тип,класс,подкласс,Марка + модель,Владелец,Область
0,Январь,2009,nfz,Transporter,C,Peugeot Boxer,Юридическое лицо,МИНСК
1,Март,2009,pkw,Hatchback,A0,Skoda Fabia,Юридическое лицо,ГРОДНЕНСКАЯ ОБЛАСТЬ
2,Ноябрь,2011,pkw,Sedan,A0,Kia Rio,Физическое лицо,ГРОДНЕНСКАЯ ОБЛАСТЬ
3,Декабрь,2011,pkw,Hatchback,A,Volkswagen Golf,Физическое лицо,ГРОДНЕНСКАЯ ОБЛАСТЬ
4,Апрель,2008,pkw,SUV,A,Mitsubishi Outlander,Физическое лицо,МИНСК
...,...,...,...,...,...,...,...,...
5994,Июнь,2011,pkw,Sedan,A,Toyota Corolla,Физическое лицо,МИНСК
5995,Август,2011,pkw,Sedan,A,Citroen C4,Физическое лицо,МИНСК
5996,Июнь,2010,pkw,Sedan,A0,Volkswagen Polo Sedan,Физическое лицо,МИНСК
5997,Февраль,2010,pkw,SUV,A,Volkswagen Tiguan,Юридическое лицо,МИНСКАЯ ОБЛАСТЬ


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5999 entries, 0 to 5998
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Месяц           5999 non-null   str  
 1   Год             5999 non-null   int64
 2   тип             5999 non-null   str  
 3   класс           5999 non-null   str  
 4   подкласс        5999 non-null   str  
 5   Марка + модель  5999 non-null   str  
 6   Владелец        5999 non-null   str  
 7   Область         5999 non-null   str  
dtypes: int64(1), str(7)
memory usage: 375.1 KB


In [5]:
df['Марка + модель'].value_counts()

Марка + модель
Volkswagen Polo Sedan    435
Renault Sandero          283
Hyundai Accent           268
Kia Rio                  242
Renault Duster           221
                        ... 
Mercedes-Benz CLA          1
Suzuki Swift               1
Hyundai Azera              1
Toyota Sequoia             1
Ford EcoSport              1
Name: count, Length: 232, dtype: int64

In [6]:
df['Область'].value_counts()

Область
МИНСК                  2899
МИНСКАЯ ОБЛАСТЬ         806
ГОМЕЛЬСКАЯ ОБЛАСТЬ      570
ВИТЕБСКАЯ ОБЛАСТЬ       559
МОГИЛЕВСКАЯ ОБЛАСТЬ     408
БРЕСТСКАЯ ОБЛАСТЬ       388
ГРОДНЕНСКАЯ ОБЛАСТЬ     369
Name: count, dtype: int64

In [7]:
df['Марка + модель'].str.split().str[0]

0          Peugeot
1            Skoda
2              Kia
3       Volkswagen
4       Mitsubishi
           ...    
5994        Toyota
5995       Citroen
5996    Volkswagen
5997    Volkswagen
5998        Toyota
Name: Марка + модель, Length: 5999, dtype: object

In [12]:
1. Вам представлены данные по продажам автомобилей в РБ. Ответьте на следующие вопросы:
     1) Какие бренды наиболее популярны?
     2) Доля Volkswagen в Гомельской области?
     3) Доля Kia Sportage в сегменте?
     4) Основные конкуренты Volkswagen Tiguan?
"Рассмотрите предоставленную статистику. Что происходило с автомобильным рынком в РБ на протяжении рассматриваемого периода? Какие внешние и внутренние факторы на это могли повлиять?
Выводы опишите в аналитической записке для руководства."
2. Постройте поквартальный прогноз продаж по всему модельному ряду VW на 2012-2013 год


SyntaxError: unterminated string literal (detected at line 6) (100748461.py, line 6)

In [13]:
# Подготовка признаков (идемпотентно)
df['Марка'] = df['Марка + модель'].str.split().str[0]
_m2q = {'Январь': 1, 'Февраль': 1, 'Март': 1, 'Апрель': 2, 'Май': 2, 'Июнь': 2,
        'Июль': 3, 'Август': 3, 'Сентябрь': 3, 'Октябрь': 4, 'Ноябрь': 4, 'Декабрь': 4}
df['Квартал'] = df['Месяц'].map(_m2q)

print('Продажи по годам:')
print(df['Год'].value_counts().sort_index())

Продажи по годам:
Год
2008     863
2009     826
2010    1501
2011    2809
Name: count, dtype: int64


In [14]:
# 1) Самые популярные бренды
brands = df['Марка'].value_counts()
print('Топ-10 брендов по продажам в РБ (2008–2011):')
print(brands.head(10))

fig = px.bar(brands.head(10).iloc[::-1], orientation='h',
             title='Топ-10 брендов по продажам в РБ (2008–2011)',
             labels={'value': 'Продажи', 'index': 'Марка'})
fig.update_layout(showlegend=False)
fig.show()

Топ-10 брендов по продажам в РБ (2008–2011):
Марка
Volkswagen    914
Renault       751
Kia           541
Hyundai       427
Nissan        402
Skoda         355
Toyota        324
Ford          223
Chevrolet     213
Mitsubishi    207
Name: count, dtype: int64


In [17]:
# 2) Доля Volkswagen в Гомельской области
gomel = df[df['Область'] == 'ГОМЕЛЬСКАЯ ОБЛАСТЬ']
vw_share = (gomel['Марка'] == 'Volkswagen').mean() * 100
print(f"Доля Volkswagen в Гомельской области: {vw_share:.1f}% "
      f"({(gomel['Марка']=='Volkswagen').sum()} из {len(gomel)}) — 2-е место после Renault")
print('\nТоп-5 брендов в Гомельской области, %:')
print((gomel['Марка'].value_counts(normalize=True).head(5) * 100).round(1))

Доля Volkswagen в Гомельской области: 15.1% (86 из 570) — 2-е место после Renault

Топ-5 брендов в Гомельской области, %:
Марка
Renault       17.4
Volkswagen    15.1
Nissan        12.3
Kia            9.3
Hyundai        6.8
Name: proportion, dtype: float64


In [16]:
# 3) Доля Kia Sportage в сегменте
suv = df[df['класс'] == 'SUV']
share_suv = (suv['Марка + модель'] == 'Kia Sportage').mean() * 100
print(f"Kia Sportage в классе SUV: {share_suv:.1f}% "
      f"({(suv['Марка + модель']=='Kia Sportage').sum()} из {len(suv)})")

suv_a = df[(df['класс'] == 'SUV') & (df['подкласс'] == 'A')]
share_a = (suv_a['Марка + модель'] == 'Kia Sportage').mean() * 100
print(f"Kia Sportage в компактных SUV (подкласс A): {share_a:.1f}% "
      f"({(suv_a['Марка + модель']=='Kia Sportage').sum()} из {len(suv_a)}) — лидер сегмента")

Kia Sportage в классе SUV: 8.6% (178 из 2063)
Kia Sportage в компактных SUV (подкласс A): 18.4% (178 из 967) — лидер сегмента


In [18]:
# 4) Основные конкуренты Volkswagen Tiguan (сегмент SUV, подкласс A)
seg = df[(df['класс'] == 'SUV') & (df['подкласс'] == 'A')]
top_seg = seg['Марка + модель'].value_counts().head(10)
print('Сегмент компактных SUV (подкласс A):')
print(top_seg)
print(f"\nВсего в сегменте: {len(seg)}. Tiguan: {(seg['Марка + модель']=='Volkswagen Tiguan').sum()} "
      f"({(seg['Марка + модель']=='Volkswagen Tiguan').mean()*100:.1f}%)")

fig = px.bar(top_seg.iloc[::-1], orientation='h',
             title='Сегмент компактных SUV (подкласс A): Tiguan и конкуренты',
             labels={'value': 'Продажи', 'index': 'Модель'})
fig.update_layout(showlegend=False)
fig.show()

Сегмент компактных SUV (подкласс A):
Марка + модель
Kia Sportage            178
Nissan Qashqai          107
Mitsubishi Outlander     76
Toyota RAV4              76
Hyundai Tucson           71
Volkswagen Tiguan        53
Mazda CX-5               34
Geely Emgrand            34
Honda CR-V               32
Chevrolet Captiva        32
Name: count, dtype: int64

Всего в сегменте: 967. Tiguan: 53 (5.5%)


## Аналитическая записка для руководства

**Динамика рынка 2008–2011** (выборка 5999 регистраций):

| Год | Продажи | Δ к пред. году |
|---|---|---|
| 2008 | 863 | — |
| 2009 | 826 | −4% |
| 2010 | 1501 | +82% |
| 2011 | 2809 | +87% |

После стагнации 2008–2009 рынок резко вырос в 2010–2011 гг., с пиком в **IV квартале 2011** (у VW Q4-2011 = 184 против Q4-2010 = 95).

**Внешние факторы:**
- **Мировой финансовый кризис 2008–2009** — провал спроса в 2009 г.
- **Валютный кризис в Беларуси 2011 г.**: девальвация рубля почти в 3 раза. Обесценивающиеся рубли население и бизнес вкладывали в товары длительного пользования, в т.ч. авто → ажиотажный спрос и пик IV кв. 2011.
- **Таможенный союз (2010–2011)**: ожидание роста ввозных пошлин и утилизационного сбора подталкивало к покупкам «впрок».
- Доступность автокредитов до середины 2011 г., затем сворачивание из-за роста ставок.

**Внутренние факторы:**
- Развитие дилерских сетей и расширение бюджетного модельного ряда (Renault, Kia, Hyundai, Geely).
- Концентрация спроса в Минске (≈48% регистраций) — платёжеспособность столицы.
- Смещение спроса в бюджетные седаны (Polo Sedan, Sandero, Accent, Rio) и SUV.

**Выводы:**
- Рост 2010–2011 в значительной мере разогнан разовым кризисным ажиотажем 2011 г., а не устойчивым улучшением платёжеспособности.
- В 2012 г. вероятна коррекция вниз: падение реальных доходов после девальвации, рост пошлин/утильсбора, исчерпание отложенного спроса.
- Поэтому прогноз, экстраполирующий тренд 2008–2011, следует трактовать как **оптимистичный сценарий**; консервативный сценарий предполагает продажи ниже.

## 2. Поквартальный прогноз продаж VW на 2012–2013 (Holt-Winters)

Метод: экспоненциальное сглаживание (Holt-Winters) с **аддитивным трендом** и **сезонностью период = 4 квартала**, обучено на 16 кварталах 2008–2011.

⚠️ Важно: обучающий ряд включает кризисный ажиотаж 2011 г., поэтому модель экстраполирует высокий тренд. Прогноз следует трактовать как **оптимистичный сценарий** — фактические продажи 2012 г. в реальности были ниже из-за коррекции рынка после девальвации.

In [ ]:
from statsmodels.tsa.holtwinters import ExponentialSmoothing

# Поквартальный ряд продаж всего модельного ряда VW (2008–2011)
vw = df[df['Марка'] == 'Volkswagen']
q = vw.groupby(['Год', 'Квартал']).size().sort_index()
idx = pd.PeriodIndex([f'{y}Q{qq}' for y, qq in q.index], freq='Q')
ts = pd.Series(q.values, index=idx, dtype=float)

# Holt-Winters: аддитивный тренд + сезонность (4 квартала)
model = ExponentialSmoothing(ts, trend='add', seasonal='add', seasonal_periods=4).fit()
fc = model.forecast(8)
fc.index = pd.period_range('2012Q1', periods=8, freq='Q')
fc = fc.round().astype(int)

print('Прогноз продаж VW по кварталам:')
print(fc)
print(f'\nИтого за год — прогноз: 2012 = {fc[:4].sum()}, 2013 = {fc[4:].sum()}')
print('Факт (все модели VW): 2010 = 237, 2011 = 376')

# График: история + прогноз
hist_df = pd.DataFrame({'Квартал': ts.index.to_timestamp(), 'Продажи': ts.values, 'Тип': 'Факт'})
pred_df = pd.DataFrame({'Квартал': fc.index.to_timestamp(), 'Продажи': fc.values, 'Тип': 'Прогноз'})
plot_df = pd.concat([hist_df, pred_df], ignore_index=True)
fig = px.line(plot_df, x='Квартал', y='Продажи', color='Тип', markers=True,
              title='VW: поквартальные продажи — факт 2008–2011 и прогноз 2012–2013')
fig.show()